tests/test_expected_protocol.py

```diff
@@ -43,6 +50,41 @@ def test_simple_protocol():
         instr.simple = 4.5
 
 
+def test_not_all_communication_used():
+    """Test whether unused communication raises an error."""
+    with raises(AssertionError) as exc:
+        with expected_protocol(BasicTestInstrument,
+                               [('VOLT?', 3.14),
+                                ('VOLT 4.5 V',),
+                                ]) as instr:
+            assert instr.simple == 3.14
+    assert str(exc.value) == "Not all messages exchanged."
```

Comment:
as above, `with pytest.raises(AssertionError, match=...` should be able to do this in one go.

Reply:
Thanks for that hint.